apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -6162,3 +6160,27 @@ sc_ws_broken_call_code_(Owner, GetVolley, _ConnPid1, _ConnPid2,
     #{<<"return_type">> := <<"error">>} = Res,
     ok.
 
+mine_micro_block_emptying_mempool_or_fail(Node) ->
+    {ok, [KeyBlock, MicroBlock]} = aecore_suite_utils:mine_micro_blocks(Node, 1),
+    true = aec_blocks:is_key_block(KeyBlock),
+    false = aec_blocks:is_key_block(MicroBlock),
+    case rpc:call(Node, aec_tx_pool, peek, [infinity]) of
+        {ok, []} -> ok;
+        {ok, [_|_] = MempoolTxs} ->
+            NewBlocks = aecore_suite_utils:flush_new_blocks(),
+            case
+                (length(NewBlocks) =:= 1) andalso aec_blocks:is_key_block(hd(NewBlocks)) andalso
+                (aec_blocks:hash_internal_representation(KeyBlock) =:= {ok, aec_blocks:prev_key_hash(MicroBlock)}) andalso
+                (aec_blocks:prev_key_hash(MicroBlock) =:= aec_blocks:prev_key_hash(hd(NewBlocks)))
+            of
+                true ->
+                    ct:fail("Key block mined shortly after micro block mined pushed the micro block out of the chain.");
```

Comment:
Sample output on console: see [this auxiliary CI job](https://circleci.com/gh/aeternity/aeternity/13429#tests/containers/1).